### ISSUE TITLE
How to do stratification in TuneWrapper

### ISSUE THREAD

**Issue by smilesun** [mlr-org/mlr#1378]
The following code is quite self explanary to my problem,  the problem is how I could tell makeTuneWrapper() only to train on some specific features of the task since some of the columns in the task has to be used as stratification columns. 

```
mTuneWrapperAsLearner = function(lrn, ps, ctrl = getMBOcontrol(30), task = getTransformedTask())
{
  inner = makeResampleDesc("CV", iters = 3, predict = "both", stratify.cols = gconf$stratifyCols )
  inner_in = makeResampleInstance(inner, task)
  lrn2 = makeTuneWrapper(lrn, resampling = inner_in, par.set = ps, control = ctrl, measures = genMeasures())
  # The problem is how to tell "makeTuneWrapper" the resampling should be make only to the features but not stratification columns? 
  return(lrn2)
}


runNestedCV = function(lrn2, feat = getFeatName(), task = getTransformedTask(), measures = genMeasures() )
{
  outer = makeResampleDesc("CV", iters = 3, predict = "both", stratify.cols = gconf$stratifyCols )
  outer_in = makeResampleInstance(outer, task)
  parallelMap::parallelStartMulticore(cpus = 3)
  r = resample(lrn2, task = subsetTask(task, features = feat), resampling = outer_in, extract = getTuneResult, measures = measures)
  parallelMap::parallelStop()
  #getTuneResult {mlr}: Returns the optimal hyperparameters and optimization path after training.
  print(r)
  print(r$extract) # this allows access to ALL tuning data, it is simply a list of TuneResult objects
  return(r) 
}

wksvm = mTuneWrapperAsLearner(makeLearner('regr.ksvm', id = 'ksvm'), ps = ps_ksvm)
rst = runNestedCV(wksvm)
```

**Comment by larskotthoff** [comment]
You could use a feature filter wrapper, but if you want to exclude some features from everything simply do so when creating the task. You can use the `blocking` argument to specify folds when creating the task.

**Comment by smilesun** [comment]
Hi, Lars, I am not sure if there is any misunderstanding about filter wrapper from my side. The point is I have to do stratification on around 20 columns which means I have to put these columns into the re-sample to do a fair distribution of those stratification factors but still in the training of learner, I can't use these stratification columns

**Comment by larskotthoff** [comment]
Then use the `blocking` argument when you create the task and omit those columns from the data.

**Comment by smilesun** [comment]
I found this in the tutorial "For example, we could include a blocking factor in the task. This would indicate that some observations "belong together" and should not be separated when splitting the data into training and test sets for resampling" I think this is to prevent stratification and only works for columns?

**Comment by larskotthoff** [comment]
Maybe I'm misunderstanding what you're trying to do, but I think this does exactly what you want. You don't get stratification directly, but you can use the stratification you have to define the blocks.

Oh and all of this is documented in the help for the respective functions.

**Comment by smilesun** [comment]
I would agree with Jakob that we make this as an open question. As Lars explained,  one could define the blocks to forbidden samples from  being split into training and validation. But if one would like all the levels of one(or 10) factor column(2) to be equally distributed in training and validation set to include different scenarios in training set, we need to use the mlr function stratification.  And we still need to tune this.
